static/js/recent_topics.js

```diff
@@ -87,7 +88,18 @@ function format_topic(topic_data) {
     const time = new XDate(last_msg.timestamp * 1000);
     const last_msg_time = timerender.last_seen_status_from_date(time);
     const is_starred = starred_messages.is_topic_starred(stream_id, topic);
-    const hidden = muting.is_topic_muted(stream_id, topic);
+    // We hide the row according to filters or if it's muted.
+    let hidden = muting.is_topic_muted(stream_id, topic);
+    const unread_count = unread.unread_topic_counter.get(stream_id, topic);
+    if (unread_count === 0 && filters.has('unread')) {
+        hidden = true;
+    }
+    if (!is_starred && filters.has('starred')) {
+        hidden = true;
+    }
+    if (!topic_data.participated && filters.has('participated')) {
+        hidden = true;
+    }
```

Comment:
You might want to extract a `should_show_row(stream_id, topic, filters)` that can just return the value used as `hidden` here; that's be more readable.

Reply:
I think it would lead to repeated computation of things like `is_starred`, `unread_count` or `participated`, or would have to returned from the extracted function.